client/src/ui/components/trading/MarketModal.tsx

```diff
@@ -59,6 +62,21 @@ export const MarketModal = () => {
 
   const { bidOffers, askOffers } = useSetMarket();
 
+  const battles = useBattlesByPosition(bank?.position || { x: 0, y: 0 });
+
+  const currentBlockTimestamp = useUIStore.getState().nextBlockTimestamp || 0;
+
+  const getStructure = useStructureByPosition();
+
+  const bankStructure = getStructure(bank?.position || { x: 0, y: 0 });
+
+  const battle = useMemo(() => {
```

Comment:
this will not be the bank battle if another battle between two armies happen on the same hex I think